executor/point_get.go

```diff
@@ -193,6 +211,11 @@ func (e *PointGetExecutor) get(ctx context.Context, key kv.Key) (val []byte, err
 		if !kv.IsErrNotFound(err) {
 			return nil, err
 		}
+		var ok bool
+		val, ok = e.ctx.GetSessionVars().TxnCtx.GetKeyInPessimisticLockCache(key)
```

Comment:
If the pessimistically locked first and the value is cached, this transaction does another read optimistically, will it get from the cache?

Reply:
No.